src/node_platform.cc

```diff
@@ -127,7 +128,7 @@ class WorkerThreadsTaskRunner::DelayedTaskScheduler {
 
     void Run() override {
       uint64_t delay_millis =
-          static_cast<uint64_t>(delay_in_seconds_ + 0.5) * 1000;
+          static_cast<uint64_t>(llround(delay_in_seconds_ + 0.5)) * 1000;
```

Comment:
I'm not sure this is the optimal solution. The idiom of `static_cast<uint64_t>(delay_in_seconds_ + 0.5)` is used here instead of `ceil`, so the change should probably be 
```suggestion
          trunc(delay_in_seconds_ + 0.5) * 1000ULL;
```
But even then we have signed/unsigned mismatch...

... So I'm not sure what _is_ the optimal solution here ...

Reply:
I think the key might be to figure out what are the constraint on `delay_in_seconds_`. Is it garnered to be whole? or positive?

Reply:
@refack Should I go with `trunc` or give this more time?

Reply:
@refack ping.

Reply:
```suggestion
          llround(delay_in_seconds_) * 1000;
```
I think this is the most correct code here.

Reply:
done.